docs/project/linux-performance-tracing.md

```diff
@@ -46,7 +46,7 @@ Follow these steps to prepare your machine to collect a performance trace.
 
 ## Collecting a Trace ##
 1. Have two shell windows available - one for controlling tracing, referred to as **[Trace]**, and one for running the application, referred to as **[App]**.
-2. **[App]** Setup the application shell - this enables tracing configuration inside of CoreCLR.
+2. **[App]** Setup the application shell - this enables tracing configuration inside of CoreCLR. This incurs a significant performance overhead (10-20%).
```

Comment:
```suggestion
2. **[App]** Setup the application shell - this enables tracing configuration inside of CoreCLR.

    > ```bash
	> export COMPlus_PerfMapEnabled=1
	> export COMPlus_EnableEventLog=1
	> ```

    > [!NOTE]
    > COMPlus_PerfMapEnabled will cause the .NET runtime to write a file containing symbolic information for jitted code to the disk. Depending on the performance of your disk and the amount of jitted code in the application this could have a significant performance overhead.
```

I'd worry the "10-20% overhead" is ambiguous (startup? throughput? steady-state memory? disk IO?). I would expect this is also highly scenario dependent. Some environments have huge code + slow disks and some have small code + fast disks. I could imagine some users will find the overheads neglible and others will find them huge. 10-20% might mislead people into thinking the situation will be better or worse than it really is for them.

Reply:
@brianrob - Brian you've probably worked the most with folks using perfcollect. Let us know if you have any suggestions on what this should say.

Reply:
I agree with the proposed suggestion from @noahfalk.  I'd be careful to quote a percentage, since it will be specific to the workload.  One thing worth calling out is that we also write data out for R2R methods that are called, so you should remove the specific callout to jitted code.  Starting in .NET 5, the jitdump file, which is enabled alongside the `perf-%pid.map` output will include R2R methods.

Reply:
I expect `COMPlus_EnableEventLog` to have relatively large perf hits if something like perfcollect runs (orders of magnitude). `COMPlus_PerfMapEnabled` should be a much much lower perf overhead, mostly concurrency + IO dominated as you say.

Reply:
I expect COMPlus_EnableEventLog to have relatively large perf hits if something like perfcollect runs (orders of magnitude). COMPlus_PerfMapEnabled should be a much much lower perf overhead, mostly concurrency + IO dominated as you say.

Reply:
I would expect EnableEventLog overhead to be dependent on what events have been enabled. I don't think we could say much about it in isolation from the rest of that configuration? In practice are we seeing signs that the perf overhead is an issue? This issue was originally filed about Perf map and not EnableEventLog so if there are also EventLog concerns we may want to open another issue to clarify what they are.

Reply:
The default `perfcollect` guidance in dev box scenarios I've seen made it almost 100 times slower. I haven't seen anyone do that in prod nonetheless.